merlin/models/tf/blocks/core/transformations.py

```diff
@@ -556,3 +556,76 @@ def call_outputs(
         targets = transform_label_to_onehot(targets, num_classes)
 
         return outputs.copy_with_updates(targets=targets)
+
+
+@tf.keras.utils.register_keras_serializable(package="merlin_models")
+class PopularityLogitsCorrection(Block):
+    """Correct the predicted logit scores based on the item frequency
+
+    This first implementation requires the user to provide
+    the item frequency table.
+
+    TODO: Use the schema to infer the path to the item frequency parquet table
+
+    Parameters:
+    ----------
+    item_frequency : Union[tf.Tensor]
+        The item frequency table
+    """
+
+    def __init__(self, item_frequency: Union[tf.Tensor, str], **kwargs):
```

Comment:
Maybe we should also have the schema as an argument, so that we retrieve the item_id cardinality from the schema and check if it matches the shape of `item_frequency`, just for defensive programming.